Makefile.generic

```diff
@@ -208,8 +207,21 @@ packages:
 		fi;\
 	done
 
+ifneq (,$(PACKAGE))
 .PHONY: package
-package: dist-build-dep dist-package dist-copy-out $(SOURCE_COPY_OUT)
+ifeq (,$(CACHE_FILE))
```

Comment:
I'd like to avoid too many conditionals if possible. And I think it is possible here. Those conditionals (on CACHE_FILE and on PACKAGE) seems to be needed only because CACHE_FILE is set by a builder plugin conditionally (if the builder plugin supports it, and if PACKAGE is set). But I think builder plugin doesn't need to be involved here. What you need is a build timestamp to compare with source files, it doesn't need to have any specific content. It would be simpler to create a build stamp file here (independent of existing .list) based on variables you already have. Bonus - it will work without any builder plugin change, which means you have it for all the package types for free.

Besides the above, I don't like `CACHE_FILE` name - it suggests some content is actually cached inside, plus it may be confused with `CACHEDIR`. `BUILD_STAMP_FILE`? 

So, it could look like this (untested):
```
BUILD_STAMP_FILE = $(ORIG_SRC)/$(OUTPUT_DIR)/$(notdir $(PACKAGE).build_stamp
$(BUILD_STAMP_FILE) : $(addprefix $(ORIG_SRC)/,$(shell scripts/changed-files $(ORIG_SRC)))
    $(info  -> Rebuilding packages for $(COMPONENT)/pkgs/$(PACKAGE_SET)-$(DIST) from $(PACKAGE))
    $(MAKE) -f $(THIS_MAKEFILE) PACKAGE=$(PACKAGE) dist-build-dep dist-package dist-copy-out $(SOURCE_COPY_OUT)
    touch "$@"

package: ...
```

Reply:
Done, but I moved the whole rebuilding logic to own Goal. This has the effect that the order of `|dist-build-dep dist-package…` Prereqs is preserved. This will avoid issues once I fill `qubes-builder` to be able to run with `make -j`